File: C#-OOP-Basics/Problems Course Instance 2018/DefiningClasses/FamilyTree/Person.cs
using System.Collections.Generic;
using System.Text;

public class Person
{
    public Person()
    {
        this.Parents = new List<Person>();
        this.Children = new List<Person>();
    }

    public Person(string firstName, string lastName, string date) :this()
    {
        this.FirstName = firstName;
        this.LastName = lastName;
        this.BirthDate = date;
    }

    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string BirthDate { get; set; }
    public List<Person> Parents { get; set; }
    public List<Person> Children { get; set; }

    public override string ToString()
    {
        StringBuilder sb = new StringBuilder();
        sb.AppendLine($"{this.FirstName} {this.LastName} {this.BirthDate}");
        sb.AppendLine("Parents:");

        foreach (var parent in this.Parents)
        {
            sb.AppendLine($"{parent.FirstName} {parent.LastName} {parent.BirthDate}");

        }

        sb.AppendLine("Children:");

        foreach (var child in this.Children)
        {
            sb.AppendLine($"{child.FirstName} {child.LastName} {child.BirthDate}");

        }

        return sb.ToString();
    }
}

File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson01DefiningClasses/Exercise/pr10_family_tree/Person.java
package Lesson01DefiningClasses.Exercise.pr10_family_tree;

import java.util.LinkedList;
import java.util.List;

public class Person {
    private String name;
    private String birthDate;
    private List<Person> parents;
    private List<Person> children;

    public Person(String name, String birthDate) {
        super();
        this.name = name;
        this.birthDate = birthDate;
        this.parents = new LinkedList<>();
        this.children = new LinkedList<>();
    }

    void addParent(Person parent) {
        this.parents.add(parent);
    }

    void addChild(Person child) {
        this.children.add(child);
    }

    public String getName() {
        return this.name;
    }

    String getBirthDate() {
        return this.birthDate;
    }

    private String getInfo() {
        return String.format("%s %s", this.name, this.birthDate);
    }

    String getFullInfo() {
        StringBuilder sb = new StringBuilder();
        sb.append(this.getInfo()).append(System.lineSeparator());
        sb.append("Parents:").append(System.lineSeparator());
        for (Person parent : this.parents) {
            sb.append(parent.getInfo()).append(System.lineSeparator());
        }
        sb.append("Children:").append(System.lineSeparator());
        for (Person child : this.children) {
            sb.append(child.getInfo()).append(System.lineSeparator());
        }
        return sb.toString();
    }
}
